include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -184,1400 +170,959 @@ class alignment_file_input_format_REMOVEME<format_sam>
+                  "over letters that model seqan3::alphabet.");
+
+    static_assert(tuple_like<remove_cvref_t<mate_type>>,
+                  "The mate object must be a std::tuple of size 3 with "
+                  "1) a std::ranges::forward_range with a value_type modelling seqan3::alphabet, "
+                  "2) a std::integral or std::optional<std::integral>, and "
+                  "3) a std::integral.");
+
+    static_assert(((std::ranges::forward_range<decltype(std::get<0>(mate))>     ||
+                    std::integral<remove_cvref_t<decltype(std::get<0>(mate))>> ||
+                    detail::is_type_specialisation_of_v<remove_cvref_t<decltype(std::get<0>(mate))>, std::optional>) &&
+                  (std::integral<remove_cvref_t<decltype(std::get<1>(mate))>> ||
+                   detail::is_type_specialisation_of_v<remove_cvref_t<decltype(std::get<1>(mate))>, std::optional>) &&
+                  std::integral<remove_cvref_t<decltype(std::get<2>(mate))>>),
+                  "The mate object must be a std::tuple of size 3 with "
+                  "1) a std::ranges::forward_range with a value_type modelling seqan3::alphabet, "
+                  "2) a std::integral or std::optional<std::integral>, and "
+                  "3) a std::integral.");
+
+    if constexpr (std::integral<remove_cvref_t<decltype(std::get<0>(mate))>> ||
+                  detail::is_type_specialisation_of_v<remove_cvref_t<decltype(std::get<0>(mate))>, std::optional>)
+        static_assert(!detail::decays_to_ignore_v<header_type>,
+                      "If you give indices as mate reference id information the header must also be present.");
+
+    static_assert(std::same_as<remove_cvref_t<tag_dict_type>, sam_tag_dictionary>,
+                  "The tag_dict object must be of type seqan3::sam_tag_dictionary.");
+
+    // ---------------------------------------------------------------------
+    // logical Requirements
+    // ---------------------------------------------------------------------
+    if constexpr (!detail::decays_to_ignore_v<header_type> &&
+                  !detail::decays_to_ignore_v<ref_id_type> &&
+                  !std::integral<std::remove_reference_t<ref_id_type>> &&
+                  !detail::is_type_specialisation_of_v<std::remove_reference_t<ref_id_type>, std::optional>)
+    {
 
-        // The rest of the header lines
-        // -------------------------------------------------------------------------------------------------------------
-        while (is_char<'@'>(*std::ranges::begin(stream_view)))
+        if (options.sam_require_header && !std::ranges::empty(ref_id))
         {
-            std::ranges::next(std::ranges::begin(stream_view));              // skip @
+            auto id_it = header.ref_dict.end();
 
-            if (is_char<'S'>(*std::ranges::begin(stream_view)))              // SQ (sequence dictionary) tag
+            if constexpr (std::ra
```

Comment:
The braces in this switch statement are redundant.

Reply:
Did this comment really correspond to line 727? I don't understand

Reply:
Some problem of Github. If you look at the actual file, my comment annotates line 1019. Also the snippet above isn't actually in the file.

Reply:
hm so we never put braces in case statements?

Reply:
We put them there if we need to, e.g. when declaring local variables.

Regular C++ rules for switch-case :wink: